Asterix/optics/phase_amplitude_functions.py

```diff
@@ -220,3 +221,410 @@ def sine_cosine_basis(Nact1D):
+        Option for the center pixel.
+        If 'p', center on the pixel dim_pp//2.
+        If 'b', center in between pixels dim_pp//2 -1 and dim_pp//2, for 'dim_pp' odd or even.
+
+    Returns
+    ------
+        apodizer_pupil : 2D array.
+            Apodizer pupil
+    """
+    if grey_pup_bin_factor > 1:
+        if not isinstance(grey_pup_bin_factor, int):
+            raise ValueError(f"grey_pup_bin_factor must be an integer, currently it is {grey_pup_bin_factor}")
+
+        if center_pos.lower() == 'p' and (dim_pp % 2 == 0 or grey_pup_bin_factor % 2 == 0):
+            raise ValueError(("if grey_pup_bin_factor>1, the pupil has to be perfectly centered:",
+                              "if center is 'p', dimpp and grey_pup_bin_factor must be odd"))
+
+        if center_pos.lower() == 'b' and (dim_pp % 2 == 1 or grey_pup_bin_factor % 2 == 1):
+            raise ValueError(("if grey_pup_bin_factor>1, the pupil has to be perfectly centered:",
+                              "if center is 'b', dimpp and grey_pup_bin_factor must be even"))
+        # we add valueError conditions because it is very hard to maintain the same centering after the
+        # rebin in all conditions
+
+        if center_pos.lower() == 'b':
+            dimpp_pup_large = (2 * int(np.ceil(prad))) * grey_pup_bin_factor
+            center_on_pixel = False
+        elif center_pos.lower() == 'p':
+            dimpp_pup_large = (2 * int(np.ceil(prad)) + 1) * grey_pup_bin_factor
+            center_on_pixel = True
+        else:
+            raise ValueError("center_pos must be 'p' (centered on pixel) or 'b' (centered in between 4 pixels)")
+
+        apodizer_pupil_large = make_apodizer(dimpp_pup_large,
+                                             grey_pup_bin_factor * prad,
+                                             apodizer_profile,
+                                             grey_pup_bin_factor=1,
+                                             center_pos=center_pos)
+        return crop_or_pad_image(
+            rebin(apodizer_pupil_large, factor=grey_pup_bin_factor, center_on_pixel=center_on_pixel), dim_pp)
+
+    else:
+        xx, yy = np.meshgrid(np.arange(dim_pp) - dim_pp // 2, np.arange(dim_pp) - dim_pp // 2)
+
+        if center_pos.lower() == 'b':
+            xx = xx + 1 / 2
+            yy = yy + 1 / 2
+        elif center_pos.lower() == 'p':
+            pass
+        else:
+            raise ValueError("center_pos must be 'p' (centered on pixel) or 'b' (centered in between 4 pixels)")
+        rr = np.hypot(yy, xx)
+
+        apodizer_pupil = np.zeros((dim_pp, dim_pp))
+        apodizer_pupil[rr <= prad] = apodizer_profile(rr[rr <= prad] / prad)
+
+        return apodizer_pupil
+
+
+def make_spider(dim_pp, starting_point, finishing_point, w_spiders, center_pos='b'):
+    """
+    Make a unique spider from starting_point to finishing point, of width w_spiders
+
+    AUTHORS : Johan Mazoyer, heavily inspired by Emiel Por in HCIpy
```

Comment:
Might be worth adding a citation of his SPIE paper then. Example of how to do that:
https://github.com/johanmazoyer/Asterix/blob/fe428422d1627e83b19bf5950d746bc682725df8/Asterix/optics/coronagraph.py#L658
With the reference further below in the docstring:
https://github.com/johanmazoyer/Asterix/blob/fe428422d1627e83b19bf5950d746bc682725df8/Asterix/optics/coronagraph.py#L663